src/MongoDB.Driver/MongoDatabaseImpl.cs

```diff
@@ -216,34 +216,48 @@ public override void CreateCollection(IClientSessionHandle session, string name,
             var genericMethodDefinition = typeof(MongoDatabaseImpl).GetTypeInfo().GetMethod("CreateCollectionHelper", BindingFlags.NonPublic | BindingFlags.Instance);
             var documentType = options.GetType().GetTypeInfo().GetGenericArguments()[0];
             var methodInfo = genericMethodDefinition.MakeGenericMethod(documentType);
-            methodInfo.Invoke(this, new object[] { session, name, options, cancellationToken });
+            try
```

Comment:
we have asserting on particular exception that is hidden over TargetInvocationException. This logic of course it's better to be rewritten on some day..